trawler/src/main/java/io/m9sweeper/trawler/framework/scans/Scanner.java

```diff
@@ -74,4 +90,93 @@ default void run() throws RuntimeException {
             throw new RuntimeException(ex);
         }
     }
-}
\ No newline at end of file
+
+    default String getRegistryAuthorization(DockerRegistry registry) throws Exception {
+        String authType = registry.getAuthType();
+        if ("ACR".equals(authType)) {
+            return getACRRegistryAuthorization(registry);
+        } else if ("GCR".equals(authType)) {
+            return getGCRRegistryAuthorization(registry);
+        } else if ("AZCR".equals(authType)) {
+            return getAZCRRegistryAuthorization(registry);
+        } else if (registry.getIsLoginRequired()) {
+            String authPartOne = "export TRIVY_USERNAME=" + escapeXsi(registry.getUsername()) + "; ";
+            String authPartTwo = "export TRIVY_PASSWORD=" + escapeXsi(registry.getPassword()) + "; ";
+            return authPartOne + authPartTwo;
+        }
+        return "";
+    }
+
+    private String escapeXsi(String authToken) {
+        return StringEscapeUtils.escapeXSI(authToken);
+    }
+
+    default String getACRRegistryAuthorization(DockerRegistry registry) {
+        String aws_account_id = TrawlerConfiguration.getInstance().dockerImageUrl().split("\\.")[0];
+
+        try {
+            Map<String, Object> authDetails = (Map<String, Object>) registry.getAuthDetails();
+
+            String region = authDetails.getOrDefault("acrDefaultRegion", "").toString();
+            String accessKey = authDetails.getOrDefault("acrAccessKey", "").toString();
+            String secretKey = authDetails.getOrDefault("acrSecretKey", "").toString();
+
+            AWSCredentials awsCredentials = new BasicAWSCredentials(accessKey, secretKey);
+            AmazonECR amazonECR = AmazonECRClientBuilder.standard()
+                    .withRegion(region)
+                    .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
+                    .build();
+
+            //Get Auth Token for Repository using its registry ID
+            GetAuthorizationTokenRequest request = new GetAuthorizationTokenRequest().withRegistryIds(aws_account_id);
```

Comment:
Like, the stuff that gets the auth token is potentially re-usable

Reply:
But idk if that's how snyk works nayways.

Reply:
Typically I wait to extract something out into the base class until the moment the 2nd implementation needs it (unless I'm absolutely sure it is truly re-usable). That avoids cluttering up the code.

Reply:
It's true that the env vars become trivy-specific, but the auth stuff isn't. I pulled it out because I was about to reuse it for Snyk :)